benches/bench.rs

```diff
@@ -360,6 +374,128 @@ fn bench_openssl(c: &mut Criterion) {
 #[cfg(not(feature = "tests-openssl"))]
 fn bench_openssl(_c: &mut Criterion) {}
 
+#[cfg(feature = "tests-glass-pumpkin")]
+fn bench_glass_pumpkin(c: &mut Criterion) {
+    use crypto_bigint::Limb;
+    use crypto_primes::hazmat::{lucas_test, AStarBase, LucasCheck, MillerRabin, Primality};
+
+    fn required_checks(bits: u32) -> usize {
+        ((bits as f64).log2() as usize) + 5
+    }
+
+    fn reference_from_rng(bit_length: u32, rng: &mut impl CryptoRngCore) -> BoxedUint {
+        loop {
+            let start = random_odd_integer::<BoxedUint>(
+                rng,
+                NonZeroU32::new(bit_length).unwrap(),
+                bit_length,
+            );
+            let sieve = Sieve::new(start.as_ref(), NonZeroU32::new(bit_length).unwrap(), false);
+            for num in sieve {
+                let odd_num = &Odd::new(num.clone()).unwrap();
+
+                let mr = MillerRabin::new(odd_num);
+                if (0..required_checks(bit_length))
+                    .any(|_| !mr.test_random_base(rng).is_probably_prime())
+                {
+                    continue;
+                }
+
+                match lucas_test(odd_num, AStarBase, LucasCheck::Strong) {
+                    Primality::Composite => continue,
+                    Primality::Prime => return num,
+                    _ => {}
+                }
+
+                return num;
+            }
+        }
+    }
+
+    fn reference_safe_from_rng(bit_length: u32, rng: &mut impl CryptoRngCore) -> BoxedUint {
```

Comment:
The word "reference" is a bit confusing to me. Perhaps just be explicit and name this "safe_prime_from_rng_like_gp"? and "prime_from_rng_like_gp" above?

Reply:
Fixed the names